packages/core/src/lib/light_push/index.ts

```diff
@@ -65,42 +63,76 @@ class LightPush extends BaseProtocol implements ILightPush {
         log("Failed to encode to protoMessage, aborting push");
         return {
           recipients,
-          error: SendError.ENCODE_FAILED
+          errors: [SendError.ENCODE_FAILED]
         };
       }
-      const query = PushRpc.createRequest(protoMessage, pubSubTopic);
-      const res = await pipe(
-        [query.encode()],
-        lp.encode,
-        stream,
-        lp.decode,
-        async (source) => await all(source)
-      );
+
+      query = PushRpc.createRequest(protoMessage, pubSubTopic);
+    } catch (error) {
+      log("Failed to encode to protoMessage", error);
+      return {
+        recipients,
+        errors: [SendError.ENCODE_FAILED]
+      };
+    }
+
+    const peers = await this.getPeers(3, true, opts?.peerId && [opts?.peerId]);
```

Comment:
Would be good to at least extract hard coded value as a const if we don't want them to be parameters just yet.